### ISSUE TITLE
v0.101.5 broke semver for CertificateRevocationList.verify_signature

### ISSUE THREAD

**Issue by cpu** [rustls/webpki#185]
https://github.com/rustls/webpki/commit/ac2faa796a3e6c0856303a1cbc2192d9c8aeadf0 was backported to the v0.101.x release in [v0.101.5](https://github.com/rustls/webpki/pull/170). This work originated in https://github.com/rustls/webpki/pull/164, but shouldn't have been backported to the 0.101.x release stream, because it changes the [`CertificateRevocationList` trait](https://github.com/rustls/webpki/blob/ac2faa796a3e6c0856303a1cbc2192d9c8aeadf0/src/crl.rs#L32C11-L48) in a semver breaking way, adding a new argument to the `verify_signature` fn. This trait and its fns are part of the public API, e.g. [in `rcgen`'s webpki tests](https://github.com/rustls/rcgen/blob/2c69f1ac43b7593fc4004206a136d5e42915f6fc/tests/webpki.rs#L532-L543) (reported by [est31](https://github.com/rustls/rcgen/pull/160#issuecomment-1728507042)).

Let's fix this in a point release and leave this note in place until then so the issue is discoverable.

**Comment by djc** [comment]
Sorry for missing this in review. 😞

**Comment by cpu** [comment]
> Sorry for missing this in review. 😞

It happens :-( I missed it reviewing the orignal work, and in the backport too.

>  Seems surprising that our semver check in CI didn't catch this?

Agreed. I thought initially the rel-0.101 branch might not have that in its CI, but it does: https://github.com/rustls/webpki/blob/7cb6c646a07f2a1899f51f728f10e925054180a8/.github/workflows/ci.yml#L230-L240

**Comment by cpu** [comment]
> Just to make sure I understand this change correctly, does v0.101.6 still contain a fix for https://github.com/advisories/GHSA-fh2r-99q2-6mmg/[RUSTSEC-2023-0053](https://rustsec.org/advisories/RUSTSEC-2023-0053.html)

That's correct: we're still limiting the number of signature validation operations that can occur during path building (including CRL signature validation operations) in v0.101.6. The only change is that the mechanism for doing that hasn't bled into the API.

**Comment by hawkw** [comment]
Okay, cool, thanks for confirming!
